src/server/search/doc_accessors.cc

```diff
@@ -86,13 +88,14 @@ SearchDocData ListPackAccessor::Serialize(const search::Schema& schema) const {
   return out;
 }
 
-string_view StringMapAccessor::GetString(string_view active_field) const {
+BaseAccessor::StringList StringMapAccessor::GetString(string_view active_field) const {
   auto it = hset_->Find(active_field);
-  return it != hset_->end() ? it->second : ""sv;
+  return it != hset_->end() ? StringList{it->second} : StringList{};
 }
 
 BaseAccessor::VectorInfo StringMapAccessor::GetVector(string_view active_field) const {
-  return search::BytesToFtVector(GetString(active_field));
+  auto strlist = GetString(active_field);
+  return strlist.empty() ? VectorInfo{} : search::BytesToFtVector(strlist.front());
```

Comment:
Nit: this is occurring twice, you could create a small function for it, your call